pubsub/cloud-client/quickstart.py

```diff
@@ -19,76 +19,110 @@
+    # TODO subscription_name = "Your Pub/Sub subscription name"
     # TODO num_messages = number of messages to test end-to-end
 
-    # Instantiates a publisher and subscriber client
+    # Instantiate a publisher and a subscriber client
     publisher = pubsub_v1.PublisherClient()
     subscriber = pubsub_v1.SubscriberClient()
 
     # The `topic_path` method creates a fully qualified identifier
     # in the form `projects/{project_id}/topics/{topic_name}`
-    topic_path = subscriber.topic_path(project_id, topic_name)
+    topic_path = publisher.topic_path(project_id, topic_name)
 
     # The `subscription_path` method creates a fully qualified identifier
     # in the form `projects/{project_id}/subscriptions/{subscription_name}`
     subscription_path = subscriber.subscription_path(
         project_id, subscription_name)
 
-    # Create the topic.
-    topic = publisher.create_topic(topic_path)
-    print('\nTopic created: {}'.format(topic.name))
+    # Create a topic.
+    try:
+        publisher.delete_topic(topic_path)
+    except NotFound:
+        pass
+    finally:
+        topic = publisher.create_topic(topic_path)
+        print('Topic created: \"{}\"'.format(topic.name))
 
     # Create a subscription.
-    subscription = subscriber.create_subscription(
-        subscription_path, topic_path)
-    print('\nSubscription created: {}\n'.format(subscription.name))
-
-    publish_begin = time.time()
+    try:
+        subscriber.delete_subscription(subscription_path)
+    except NotFound:
+        pass
+    finally:
+        subscription = subscriber.create_subscription(
+            subscription_path, topic_path)
+        print('Subscription created: \"{}\"'.format(subscription.name))
+
+    # `data` is roughly 10 Kb.
+    data = 'x' * 9600
+    # `data` must be a bytestring.
+    data = data.encode('utf-8')
+    # Initialize an empty dictionary to track messages.
+    tracker = dict()
+    pubsub_time = 0.0
 
     # Publish messages.
-    for n in range(num_messages):
-        data = u'Message number {}'.format(n)
-        # Data must be a bytestring
-        data = data.encode('utf-8')
-        # When you publish a message, the client returns a future.
+    for i in range(num_messages):
+        # When we publish a message, the client returns a future.
         future = publisher.publish(topic_path, data=data)
-        print('Published {} of message ID {}.'.format(data, future.result()))
 
-    publish_time = time.time() - publish_begin
+        tracker.update({i: {'index': i,
+                            'pubtime': time.time(),
+                            'subtime': None}})
+
+        # `future.result()` blocks and returns a unique message ID per message.
+        tracker[future.result()] = tracker.pop(i)
 
-    messages = set()
+    print('\nPublished all messages.')
 
     def callback(message):
-        print('Received message: {}'.format(message))
         # Unacknowledged messages will be sent again.
```

Comment:
I think this is something that is clear from the ack() call -- no need to comment it.   I would change the function name here, though, to make it more descriptive: "process_message_callback_function"

Reply:
... particularly because this is between the publish and subscribe parts. I don't know if I'm still reading about publish or not.